### PR TITLE
Fix empty array for FlatParamsEncoder {key: []} -> "key="

### PR DESCRIPTION
**Pull request by mrexox**
## Description
For an empty array in params hash should not ignore the key.
- [x] Test

## Additional Notes
The problem is: **each on empty array does not yield**.

### CODE REVIEW

File: lib/faraday.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #   conn.get '/'
 #
 module Faraday
-  VERSION = '0.16.0'
+  VERSION = '0.16.1'
```
Comment (olleolleolle) [inline lib/faraday.rb]: Please let the maintenance team do the update to this file.

Can you undo this change?

Comment (mrexox) [reply lib/faraday.rb]: Yep!

**Comment by mrexox** [comment]
What about CI? Seems that it is broken just for ruby `2.3.x`. Should I add a fix for it?

**Comment by olleolleolle** [comment]
@mrexox If you can create a separate PR fixing the 2.3 - or if the 2.3 is truly inaccessible, removing it from GitHub actions - that'd be super. 

Don't fix it in this focused PR. This one is good as it is.

**Comment by olleolleolle** [comment]
According to https://github.com/actions/setup-ruby/blob/master/action.yml 2.4 seems to be lower bound.

**Comment by mrexox** [comment]
@olleolleolle, done with it: https://github.com/lostisland/faraday/pull/1085

**Comment by mrexox** [comment]
Can this PR be merged?

**Comment by technoweenie** [comment]
I agree with this change, because it matches how an empty string is decoded.

```irb
>> Faraday::FlatParamsEncoder.encode(a: [])
=> ""
>> Faraday::FlatParamsEncoder.encode(a: '')
=> "a="
```

In my opinion, if you don't want the encoded url to have an empty value (`a=`), then don't send an empty value (`a: []` or `a: ''`). But if empty values are being encoded, they should behave roughly the same.
